mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -84,13 +90,21 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
             Ok(part)
         }
     }
+
+    pub fn downloaded(&self) -> usize {
```

Comment:
Maybe instead implement `available()`, the number of bytes currently in the queue. Saves callers from having to do the math themselves, since "how many bytes can I read right now" is the question they care about.

Reply:
I'm not sure `PartQueue` has enough context for that, but `RequestTask` could do the math and expose `available()` (or equivalent in terms of absolute offset/range, which I think would be clearer to use in prefetch).